components/lib/crash/src/main/java/mozilla/components/lib/crash/handler/CrashHandlerService.kt

```diff
@@ -4,42 +4,60 @@
 
 package mozilla.components.lib.crash.handler
 
-import android.app.IntentService
+import android.app.Service
 import android.content.Intent
-import mozilla.components.lib.crash.CrashReporter
+import android.os.Build
+import android.os.IBinder
+import androidx.core.app.NotificationCompat
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.launch
 import mozilla.components.lib.crash.Crash
+import mozilla.components.lib.crash.CrashReporter
+import mozilla.components.lib.crash.R
+import mozilla.components.lib.crash.notification.CrashNotification
+import mozilla.components.support.base.ids.SharedIdsHelper
 
-private const val WORKER_THREAD_NAME = "CrashHandlerService"
+private const val NOTIFICATION_TAG = "mozac.lib.crash.handlecrash"
 
 /**
  * Service receiving native code crashes (from GeckoView).
  */
-class CrashHandlerService : IntentService(WORKER_THREAD_NAME) {
-    private val logger by lazy { CrashReporter
-        .requireInstance
-        .logger
-    }
-
-    public override fun onHandleIntent(intent: Intent?) {
-        if (intent == null) {
-            return
+class CrashHandlerService : Service() {
+    private val crashReporter: CrashReporter by lazy { CrashReporter.requireInstance }
+
+    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
+        crashReporter.logger.error("CrashHandlerService received native code crash")
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val channel = CrashNotification.ensureChannelExists(this)
+            val notification = NotificationCompat.Builder(this, channel)
+                .setContentTitle(getString(R.string.mozac_lib_crash_dialog_title,
+                    crashReporter.promptConfiguration.organizationName))
```

Comment:
I using the "crash dialog title" here the right choice?

Reply:
I think so.  The name of the string is misleading.  This is actually the string we use for the crash notification.  And since this foreground service notification is not visible to the user, (The foreground service ends quick enough that no notification is shown) I chose this to match the crash notification that we put up for the user.